src/uu/pgrep/src/process.rs

```diff
@@ -302,14 +323,18 @@ impl ProcessInformation {
 
         let path = PathBuf::from(format!("/proc/{}/fd", self.pid));
 
-        let result = Rc::new(
-            fs::read_dir(path)?
-                .flatten()
-                .filter(|it| it.path().is_symlink())
-                .flat_map(|it| fs::read_link(it.path()))
-                .flat_map(TerminalType::try_from)
-                .collect::<HashSet<_>>(),
-        );
+        let mut result = fs::read_dir(path)?
```

Comment:
I think you have to handle a permission error here and use a `TerminalType::Unknown` in such a case.